pkg/envoy/types.go

```diff
@@ -83,3 +83,14 @@ const (
 	// The local cluster refers to the cluster corresponding to the service the proxy is fronting, accessible over localhost by the proxy.
 	localClusterSuffix = "-local"
 )
+
+// ProxyKind is the type used to define the proxy's kind
+type ProxyKind string
+
+const (
+	// ProxyKindSidecar implies the proxy is a sidecar
+	ProxyKindSidecar ProxyKind = "sidecar"
```

Comment:
```suggestion
	Sidecar ProxyKind = "sidecar"
```

Since the type is already ProxyKind

Reply:
Maybe `KindSidecar` works best!